### PR TITLE
[FAU-409] Switch to @wordpress/scripts

### PR DESCRIPTION
**Pull request by zhyian**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Chore

**What is the current behavior?** (You can also link to an open issue here)
https://inpsyde.atlassian.net/browse/FAU-409

**What is the new behavior (if this is a feature change)?**
 - Asset compilation has been updated to use `@wordpress/scripts`.
 - Replaced deprecated quality assurance assets workflows.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Other information**:

### CODE REVIEW

**Review by o-samaras — APPROVED** [review]
LGTM! Thanks

File: .github/workflows/front-office-qa.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            - '**.tsx'
+            - '**.md'
+            - '**.scss'
+            - 'front-office-qa.yml'
+jobs:
+    wp-scripts-lint:
+        uses: inpsyde/reusable-workflows/.github/workflows/wp-scripts-lint.yml@main
+        secrets:
+            GITHUB_USER_EMAIL: ${{ secrets.DEPLOYBOT_EMAIL }}
+            GITHUB_USER_NAME: ${{ secrets.DEPLOYBOT_USER }}
```
Comment (amiut) [inline .github/workflows/front-office-qa.yml:17]: I'm not sure where exactly these are used, as far as I can see there is no git operations in the workflow, does it make sense to omit them, they are optional inputs anyway.

**Review by amiut — APPROVED** [review]
Thanks for taking care of it, LGTM, left a small note but nothing that prevents me from approving

File: resources/svg/README.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
-### Usage
+# Usage
 
 This folder holds raw icons and generates a `sprite.svg` files out of them, If you are using an icon directly place it under `images` folder, as the files in this folder won't be available when they are built. (only `sprite.svg` file will be available after build).
```
Comment (tyrann0us) [inline resources/svg/README.md:3]: Let's improve the wording according to the Boy Scout rule.
```suggestion
This folder holds raw icons and generates `sprite.svg` files from them. If you use an icon directly, place it in the `images/` folder, as the files in this folder won't be available when they are built. Only the `sprite.svg` file will be available after the build.
```

**Review by tyrann0us — COMMENTED** [review]
Thank you for working on this! Left some minor suggestions.

File: resources/scss/components/_single-degree-program.scss [hunk 3]
```diff
@@ -1,38 +1,44 @@
-@use '../abstracts/config';
-@use '../base/base';
+// stylelint-disable no-invalid-position-at-import-rule
```
[the remark this answers is not in the document]
Comment (zhyian) [reply resources/scss/components/_single-degree-program.scss:1]: I thought about it, and even started, but it takes a long time. And in general in this case importing rules makes sense, so I suppressed it, as in the previous implementation.

File: package.json [hunk 4]
```diff
@@ -75,16 +35,21 @@
...
-    "dev": "encore dev",
-    "js:precommit": "eslint --fix",
-    "lint": "eslint ./resources --ext .js,.jsx,.ts,.tsx",
-    "lint-sass": "stylelint 'resources/scss/**/*.scss'",
-    "lint-sass:fix": "stylelint 'resources/scss/**/*.scss' --fix",
-    "lint:fix": "eslint ./resources --ext .js,.jsx,.ts,.tsx --quiet --fix",
-    "test": "jest --passWithNoTests",
-    "watch": "encore dev --profile --watch",
-    "check-types": "tsc --noEmit -project ./tsconfig.eslint.json --skipLibCheck"
+  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (zhyian) [reply package.json:38]: To be honest, I didn't check the dependencies that are not needed for building assets. But now that I looked, some of them seemed unnecessary, so I got rid of them.

**Review by tyrann0us — APPROVED** [review]
Thanks for incorporating the changes and bringing `wp-scripts` to this package! Code-wise LGTM. Thanks! 🙏🏽

**Pull request merged**